scripts/lint-sh.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+find . -not \( -path './*.git/*' -prune \) -not \( -path './package/*' -prune \) -not \( -path './scripts/*' -prune \) -type f -exec bash -c 'if echo $1 | grep -qE ".*\.sh$" || head -n1 $1 | grep -qE "^#.*(sh|bash|dash|ksh)$"; then shellcheck -f gcc $1; fi' _ {} \;
+find package -type f -exec bash -c 'if echo $1 | grep -qE ".*\.sh$" || head -n1 $1 | grep -qE "^#.*(sh|bash|dash|ksh)$"; then shellcheck -x -f gcc -s sh -eSC2039,SC1091,SC2155,SC2034 $1; fi' _ {} \;
```

Comment:
- Use of unquoted `$1` (twice)
- Use of incorrectly quoted grep pattern (unescaped uses single backslash and `$` inside of `""`)
- We don't use dash or ksh, ever.
- Extremely convoluted code. How about:
```sh
find ... | while read -r file; do
...
done
```